common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -127,8 +128,7 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
         );
         String requestBody = ObjectMapper.serializeObjectToFormUrlEncoded(request);
         Map<String, String> headers = new TreeMap<>();
-        String correlationId = UUID.randomUUID().toString();
-        headers.put("client-request-id", correlationId);
+        headers.put("client-request-id", DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
```

Comment:
I stole this change to use in https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/285 for a `0.0.7` hotfix